<launch>
  <arg name="input_cloud" default="/camera/depth/color/points"/>
  <arg name="output_cloud" default="/filtered_points"/>
  <arg name="urdf_xacro" default="$(find xarm_description)/urdf/xarm_device.urdf.xacro"/>

  <!-- xacro から robot_description をロード -->
  <param name="robot_description"
         command="$(find xacro)/xacro $(arg urdf_xacro)" />

  <!-- self filter -->
  <node pkg="robot_self_filter" type="self_filter" name="self_filter" output="screen">
    <param name="sensor_frame" value="camera_color_optical_frame"/>
    <param name="padding" value="0.04"/>
    <param name="scale" value="1.0"/>
    <remap from="cloud_in" to="$(arg input_cloud)"/>
    <remap from="cloud_out" to="$(arg output_cloud)"/>
    <rosparam param="self_see_links">
        - {name: link_base}
        - {name: link1}
        - {name: link2}
        - {name: link3}
        - {name: link4}
        - {name: link5}
        - {name: link6}
    </rosparam>
  </node>
</launch>
